x-pack/plugins/apm/common/connections.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { AgentName } from '../typings/es_schemas/ui/fields/agent';
+
+export enum NodeType {
+  service = 'service',
+  backend = 'backend',
+}
+
+interface NodeBase {
+  id: string;
+}
+
+export interface ServiceNode extends NodeBase {
+  type: NodeType.service;
+  serviceName: string;
+  agentName: AgentName;
+  environment: string;
+}
+
+export interface BackendNode extends NodeBase {
+  type: NodeType.backend;
+  backendName: string;
+  spanType: string;
+  spanSubtype: string;
+}
+
+export type Node = ServiceNode | BackendNode;
+
+export interface ConnectionMetricItem {
+  location: Node;
+  metrics: {
+    latency: {
+      value: number | null;
+      timeseries: Array<{ x: number; y: number | null }>;
```

Comment:
Can you use [Coordinate](https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/typings/timeseries.ts#L18:L18)?

Reply:
sure thing